src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -177,8 +221,18 @@ public void Save(ModelSaveContext ctx)
             ctx.SetVersionInfo(GetVersionInfo());
 
             ctx.SaveBinaryStream("OnnxModel", w => { w.WriteByteArray(Model.ToByteArray()); });
-            ctx.SaveNonEmptyString(_args.InputColumn);
-            ctx.SaveNonEmptyString(_args.OutputColumn);
+            //ctx.SaveNonEmptyString(_args.InputColumn);
+            //ctx.SaveNonEmptyString(_args.OutputColumn);
+
+            _host.AssertNonEmpty(Inputs);
+            ctx.Writer.Write(Inputs.Length);
+            foreach (var colName in Inputs)
+                ctx.SaveNonEmptyString(colName);
+
+            _host.AssertNonEmpty(Outputs);
```

Comment:
>AssertNonEmpty [](start = 18, length = 14)

CheckNonEmpty, public method should use Check instead of assert

Reply:
good catch

---
In reply to: [233267362](https://github.com/dotnet/machinelearning/pull/1586#discussion_r233267362) [](ancestors = 233267362)